### PR TITLE
simplify the PayloadSyntaxParser usage

### PR DESCRIPTION
**Pull request by shadeofblue**

### CODE REVIEW

File: golem/resources/allocation/allocation.py [hunk 1]
```diff
@@ -89,13 +89,10 @@ async def create(cls, node: "GolemNode", data: models.Allocation) -> "Allocation
...
-    ) -> Tuple[Properties, Constraints]:
+    async def get_properties_and_constraints_for_demand(self) -> Tuple[Properties, Constraints]:
         data = await self.api.get_demand_decorations([self.id])
-
         properties = Properties({prop.key: prop.value for prop in data.properties})
-
-        constraints = Constraints([parser.parse_constraints(c) for c in data.constraints])
-
+        constraints = Constraints(
+            [PayloadSyntaxParser().parse_constraints(c) for c in data.constraints]
```
Comment (approxit) [inline golem/resources/allocation/allocation.py:96]: Creating new instance each time, will take significant amount of time, as textx needs to parse `.tx` for its internal state.

File: golem/payload/parsers/__main__.py
Comment (approxit) [inline golem/payload/parsers/__main__.py:1]: Move `__main__` contents just to `golem.payload.parser.py`.

File: golem/payload/parsers/syntax.tx
Comment (approxit) [inline golem/payload/parsers/syntax.tx:1]: Move contents just to `golem/payload/parser.tx`

**Review by approxit — CHANGES_REQUESTED** [review]

File: golem/resources/allocation/allocation.py (hunk 1 above)
Comment (shadeofblue) [reply golem/resources/allocation/allocation.py]: I doubt it's "significant" in this particular context but okay, moving the instantiation to the constructor above...

**Review by approxit — APPROVED** [review]

**Pull request merged**